build_scripts/ubuntu_sim_ros_melodic.sh

```diff
@@ -0,0 +1,104 @@
+## It installs the common dependencies for all targets (including Qt Creator)
+##
+## Installs:
+## - Common dependencies libraries and tools as defined in `ubuntu_sim_common_deps.sh`
+## - ROS Melodic (including Gazebo9)
+## - MAVROS
+
+echo "Downloading dependent script 'ubuntu_sim_common_deps.sh'"
+# Source the ubuntu_sim_common_deps.sh script directly from github
+common_deps=$(wget https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim_common_deps.sh -O -)
+wget_return_code=$?
+# If there was an error downloading the dependent script, we must warn the user and exit at this point.
+if [[ $wget_return_code -ne 0 ]]; then echo "Error downloading 'ubuntu_sim_common_deps.sh'. Sorry but I cannot proceed further :("; exit 1; fi
+# Otherwise source the downloaded script.
+. <(echo "${common_deps}")
+
+# ROS Melodic
+## Gazebo simulator dependencies
+sudo apt-get install protobuf-compiler libeigen3-dev libopencv-dev -y
+
+## ROS Gazebo: http://wiki.ros.org/melodic/Installation/Ubuntu
+## Setup keys
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+## For keyserver connection problems substitute hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 above.
+sudo apt-get update
+## Get ROS/Gazebo
+sudo apt install ros-melodic-desktop-full -y
+## Initialize rosdep
+sudo rosdep init
+rosdep update
+## Setup environment variables
+rossource="source /opt/ros/melodic/setup.bash"
+if grep -Fxq "$rossource" ~/.bashrc; then echo ROS setup.bash already in .bashrc;
+else echo "$rossource" >> ~/.bashrc; fi
+eval $rossource
+
+## Install rosinstall and other dependencies
+sudo apt install python-rosinstall python-rosinstall-generator python-wstool build-essential -y
+
+
+
+# MAVROS: https://dev.px4.io/en/ros/mavros_installation.html
+## Install dependencies
+sudo apt-get install python-catkin-tools python-rosinstall-generator -y
+
+## Create catkin workspace
+mkdir -p ~/catkin_ws/src
+cd ~/catkin_ws
+catkin init
+wstool init src
+
+
+## Install MAVLink
+###we use the Kinetic reference for all ROS distros as it's not distro-specific and up to date
+rosinstall_generator --rosdistro kinetic mavlink | tee /tmp/mavros.rosinstall
+
+## Build MAVROS
+### Get source (upstream - released)
+rosinstall_generator --upstream mavros | tee -a /tmp/mavros.rosinstall
+
+### Setup workspace & install deps
+wstool merge -t src /tmp/mavros.rosinstall
+wstool update -t src
+if ! rosdep install --from-paths src --ignore-src -y; then
+    # (Use echo to trim leading/trailing whitespaces from the unsupported OS name
+    unsupported_os=$(echo $(rosdep db 2>&1| grep Unsupported | awk -F: '{print $2}'))
+    rosdep install --from-paths src --ignore-src --rosdistro melodic -y --os ubuntu:bionic
+fi
+
+if [[ ! -z $unsupported_os ]]; then
```

Comment:
FYI, this tests for version match on OS based on what ROS itself says. Do you think we should kill installation at this point rather than allowing it to continue?

Reply:
It should be killed.

Reply:
Tried this, but first attempt did not kill this as expected. I'm going to merge this now and look at it again later.

Reply:
@TSC21 I think that by the time you get to this point on Ubuntu 16.04 things have already failed pretty badly. I'm thinking perhaps we exit based on a xenial check in the first line?

Reply:
I am ok with that

Reply:
Great. Done.